AK/RedBlackTree.h

```diff
@@ -301,91 +279,86 @@ class BaseRedBlackTree {
                     sibling->color = Color::Red;
                     node = parent;
                 } else {
-                    if (!sibling->right_child || sibling->right_child->color == Color::Black) {
-                        sibling->left_child->color = Color::Black; // null check?
+                    if (!sibling->right_child() || sibling->right_child()->color == Color::Black) {
+                        sibling->left_child()->color = Color::Black; // null check?
                         sibling->color = Color::Red;
                         rotate_right(sibling);
-                        sibling = parent->right_child;
+                        sibling = parent->right_child();
                     }
                     sibling->color = parent->color;
                     parent->color = Color::Black;
-                    sibling->right_child->color = Color::Black; // null check?
+                    sibling->right_child()->color = Color::Black; // null check?
                     rotate_left(parent);
                     node = m_root; // fixed
                 }
             } else {
-                auto* sibling = parent->left_child;
+                auto* sibling = parent->left_child();
                 if (sibling->color == Color::Red) {
                     sibling->color = Color::Black;
                     parent->color = Color::Red;
                     rotate_right(parent);
-                    sibling = parent->left_child;
+                    sibling = parent->left_child();
                 }
-                if ((!sibling->left_child || sibling->left_child->color == Color::Black) && (!sibling->right_child || sibling->right_child->color == Color::Black)) {
+                if ((!sibling->left_child() || sibling->left_child()->color == Color::Black) && (!sibling->right_child() || sibling->right_child()->color == Color::Black)) {
                     sibling->color = Color::Red;
                     node = parent;
                 } else {
-                    if (!sibling->left_child || sibling->left_child->color == Color::Black) {
-                        sibling->right_child->color = Color::Black; // null check?
+                    if (!sibling->left_child() || sibling->left_child()->color == Color::Black) {
+                        sibling->right_child()->color = Color::Black; // null check?
                         sibling->color = Color::Red;
                         rotate_left(sibling);
-                        sibling = parent->left_child;
+                        sibling = parent->left_child();
                     }
                     sibling->color = parent->color;
                     parent->color = Color::Black;
-                    sibling->left_child->color = Color::Black; // null check?
+                    sibling->left_child()->color = Color::Black; // null check?
                     rotate_right(parent);
                     node = m_root; // fixed
                 }
             }
```

Comment:
If were going for code deduplication, these 2 branches are symmteric, the only differences being left and right rotation being flipped, as well as left and right children are flipped, eliminating half the branches will result in performance uplift

Reply:
@IdanHo Agreed. I just didn't make it that far yet. Thanks for pointing out this additional spot.

Reply:
No problem, note that simply extracting and templating the branch will only deduplicate code, not make it more performant, in order to reduce executed branches the code has to be written once, but act upon the nodes array based on the direction that is computed at the start